.github/workflows/publish.yml

```diff
@@ -0,0 +1,51 @@
+name: Publish
+
+on:
+  workflow_dispatch:
+    inputs:
+      package:
+        description: The package to release, this should be the path segment after packages/, without the @sapphire/ prefix
+        required: true
+
+jobs:
+  PublishPackage:
+    name: Publish @sapphire/${{ github.event.inputs.package}}
+    runs-on: ubuntu-latest
+    if: github.repository_owner == 'sapphiredev'
+    steps:
+      - name: Checkout Project
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+          ref: main
+          token: ${{ secrets.SKYRA_TOKEN }}
+      - name: Use Node.js v20
+        uses: actions/setup-node@v4
+        with:
+          node-version: 20
+          cache: yarn
+          registry-url: https://registry.yarnpkg.com/
+      - name: Install Dependencies
+        run: yarn --immutable
+      - name: Configure Git
+        run: |
+          git remote set-url origin "https://${GITHUB_TOKEN}:x-oauth-basic@github.com/${GITHUB_REPOSITORY}.git"
+          git config --local user.email "${GITHUB_EMAIL}"
+          git config --local user.name "${GITHUB_USER}"
+        env:
+          GITHUB_USER: github-actions[bot]
+          GITHUB_EMAIL: 41898282+github-actions[bot]@users.noreply.github.com
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+      - name: Bump Versions and make release
+        run: yarn bump
+        working-directory: packages/${{ github.event.inputs.package }}
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+      - name: Publish to NPM
+        working-directory: packages/${{ github.event.inputs.package }}
+        run: |
+          yarn config set npmAuthToken ${NODE_AUTH_TOKEN}
+          yarn config set npmPublishRegistry "https://registry.yarnpkg.com"
+          yarn npm publish
```

Comment:
Lets get provenances on too

Reply:
No can do, requires https://github.com/yarnpkg/berry/issues/5430

Reply:
And before you say "then use npm publish instead", no can do either because it would break dependencies such as `"@sapphire/utilities": "workspace:^"`